### PR TITLE
Qt/Bugfix: fix handling default wallet with no name

### PR DESCRIPTION
**Pull request by jonasschnelli**
If one loads a wallet via RPC (`loadwallet w2`), then select w2, select back to the default wallet (which is an empty string), that default wallet cannot be access through the RPC console because the current code only points to the wallet endpoint if the wallet name is not empty.

This is a quick fix that reenables accessing the default wallet in case an additional wallet has been loaded.

Using "" for the default wallet may not be ideal in other cases and it may make more sense to change it at a deeper level (wallet.cpp). See discussion here which where the reasons for the current behaviour in master:
https://github.com/bitcoin/bitcoin/pull/11687#issuecomment-370862718

@jnewbery @promag @ryanofsky

### CODE REVIEW

**Comment by promag** [comment]
utACK 3033bf9, it is now possible to use the *default wallet* in the console.

Please see fe9becaff as an alternative and smaller fix.

File: src/qt/rpcconsole.cpp [hunk 1]
```diff
@@ -425,7 +425,7 @@ void RPCExecutor::request(const QString &command, const QString &walletID)
             return;
         }
         std::string wallet_id = walletID.toStdString();
-        if(!RPCConsole::RPCExecuteCommandLine(m_node, result, executableCommand, nullptr, &wallet_id))
+        if(!RPCConsole::RPCExecuteCommandLine(m_node, result, executableCommand, nullptr, (use_wallet) ? &wallet_id : nullptr))
```
Comment (ryanofsky) [inline src/qt/rpcconsole.cpp]: parentheses around use_wallet seem odd

**Review by ryanofsky — COMMENTED** [review]
utACK for both 3033bf9c5354f710d5f18fbcdb3a48ec16fe98ff and fe9becaff77353df14da721086b394f418ba5f03. Promag's fix in fe9becaff77353df14da721086b394f418ba5f03 seems simpler and more elegant though, so maybe it would be good to reset this PR to that change and merge it.

**Comment by jonasschnelli** [comment]
Switch to https://github.com/bitcoin/bitcoin/commit/fe9becaff77353df14da721086b394f418ba5f03 which is the better approach... I have though similar to std::string there is no concept of `nullness` for QStrings.

**Pull request merged**